deployment-scripts/jobs-deployment/flink_clients.py

```diff
@@ -1,11 +1,37 @@
 import logging
 from abc import ABC, abstractmethod
 from typing import Any, Dict, List
+import time
 
 from cmd_utils import run_cmd
 
 
 class FlinkCli(ABC):
+    JOB_RUNNING_CHECK_RETRIES_COUNT = 5
+    JOB_RUNNING_CHECK_RETRIES_TIMEOUT = 0.2
+
+    def ensure_job_is_running(self, job_name: str) -> None:
+        """
+        Makes sure that a Flink job has status CREATED or RUNNING shortly after being run.
+        :param job_name: Flink job name
+        """
+        for check_index in range(self.JOB_RUNNING_CHECK_RETRIES_COUNT):
+            logging.info(f"Checking the state of the job: " + str(check_index))
+            status = self._get_job_status(job_name)
+            if not (status == "RUNNING" or status == "CREATED"):
+                raise RuntimeError(f"Unexpected job state. Recent status {status}.")
+            time.sleep(self.JOB_RUNNING_CHECK_RETRIES_TIMEOUT)
+        pass
+
+    @abstractmethod
+    def _get_job_status(self, job_name: str) -> str:
+        """
+        Gets the status of a Flink job. Executes `flink list` command and truncates output to retrieve the status.
```

Comment:
There is no need to explain how it's done: `Executes flink list command and truncates output to retrieve the status.`.
In fact it should not - the client can implement it in many different ways and from interface perspective it does not matter.

Reply:
Indeed, removed the `Executes flink ...` comment